### PR TITLE
Refactor codebase

### PR DESCRIPTION
**Pull request by HillaShx**

### CODE REVIEW

File: src/entities.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+    def set_fruit(self, ftype: str, value) -> None:
+        self.__dict__[ftype] = value
+
+    def get_all(self) -> dict:
+        return self.__dict__
+
+
+class Fruit:
+    def __init__(self, ftype: str, img_path: Path,):
```
Comment (HillaShx) [inline src/entities.py]: suggesting to change ftype to fruit_type

Comment (HillaShx) [reply src/entities.py]: remove comma from Path

File: src/entities.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+        self.ftype = ftype
+        self.img = pygame.image.load(img_path)
+        self.x = random.randint(100, 500)
+        self.y = 800
```
Comment (HillaShx) [inline src/entities.py:58]: change 800 to game_height

File: src/entities.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+                value.speed_y += (g * value.t)
+                value.t += 1
+
+                if value.get_y() <= 800:
+                    self.surface.blit(value.get_img(), value.get_position())
+                else:
+                    Fruit.generate_random_fruit(self.fruit_collection, key)
+
+                current_position = self.cursor.get_current_position()
+                self.cursor.draw(self.surface)
```
Comment (HillaShx) [inline src/entities.py:188]: try moving draw outside throw_fruit

File: src/game_logic.py [hunk 4]
```diff
@@ -1,110 +1,9 @@
...
 
-def is_hit(current_position, value):
-    return all([
-        not value['hit'],
-        current_position[0] > value['x'],
-        current_position[0] < value['x'] + 60,
-        current_position[1] > value['y'],
-        current_position[1] < value['y'] + 60,
-    ])
+    game.init_game()
```
Comment (jackbu131) [inline src/game_logic.py:9]:
```suggestion
    game.init_game()
    game. run_game(....)
```

File: src/entities.py [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+            self.surface.fill(WHITE)
+            self.surface.blit(self.score_text, (0, 0))
+            self.throw_fruit()
+
+            pygame.display.update()
+            CLOCK.tick(FPS)
+
+            for event in pygame.event.get():
+                if event.type == pygame.QUIT:
+                    sys.exit()
```
Comment (jackbu131) [inline src/entities.py:172]:
```suggestion
def run_game(....)
        while self.keep_going:
            self.surface.fill(WHITE)
            self.surface.blit(self.score_text, (0, 0))
            self.throw_fruit()

            pygame.display.update()
            CLOCK.tick(FPS)

            for event in pygame.event.get():
                if event.type == pygame.QUIT:
                    sys.exit()
```

**Pull request merged**